### PR TITLE
Fix 4.1.0 issue: `System.InvalidOperationException`: `Sequence contains more than one matching element` when `--trace` is provided

### PR DESCRIPTION
**Pull request by shaopeng-gh**
## Issue:
Found a bug In 4.1.0 , if `--trace` is provided will have unhandled exception.
The reason is new new pipeline change has SDK traces changed to trace, because we overwrite in BinSkim, we need to change the name ass well.

This is the full error:
image

### CODE REVIEW

File: src/BinSkim.Driver/MultithreadedAnalyzeCommand.cs [hunk 1]
```diff
@@ -171,7 +171,7 @@ public override int Run(AnalyzeOptions analyzeOptions)
         {
             Stopwatch stopwatch = null;
 
-            if (analyzeOptions.Traces.Where(s => s == "ScanTime").Any())
+            if (analyzeOptions.Trace.Where(s => s == "ScanTime").Any())
```
Comment (michaelcfanning) [inline src/BinSkim.Driver/MultithreadedAnalyzeCommand.cs:174]: You should be using the Traces property in the base class.

[https://github.com/microsoft/sarif-sdk/blob/140d730afd838e7e168985bfda38edd4ead7d297/src/Sarif.Driver/Sdk/AnalyzeOptionsBase.cs#L86](https://github.com/microsoft/sarif-sdk/blob/140d730afd838e7e168985bfda38edd4ead7d297/src/Sarif.Driver/Sdk/AnalyzeOptionsBase.cs#L86)

File: src/Test.FunctionalTests.BinSkim.Driver/BaselineTests.cs [hunk 2]
```diff
@@ -438,7 +438,7 @@ private SarifLog RunRules(StringBuilder sb, string inputFileName, Sarif.SarifVer
                 ConfigurationFilePath = "default",
                 SarifOutputVersion = Sarif.SarifVersion.Current,
                 TargetFileSpecifiers = new string[] { inputFileName },
-                Traces = Array.Empty<string>(),
+                Trace = Array.Empty<string>(),
```
Comment (michaelcfanning) [inline src/Test.FunctionalTests.BinSkim.Driver/BaselineTests.cs:441]: >Trace [](http://example.com/codeflow?start=16&length=5)

Why is this required? Can you remove it?

**Review by michaelcfanning — CHANGES_REQUESTED** [review]
:clock1:

File: src/BinSkim.Driver/AnalyzeOptions.cs [hunk 3]
```diff
@@ -21,7 +21,7 @@ public class AnalyzeOptions : AnalyzeOptionsBase
                        "(escape semicolon with backslash in Unix-like OS), that " +
                        "should be emitted to the console and log file (if appropriate). " +
                        "Valid values: PdbLoad.")]
```
[the remark this answers is not in the document]
Comment (shaopeng-gh) [reply src/BinSkim.Driver/AnalyzeOptions.cs]: Yes! found a way and fixed!

File: src/Test.FunctionalTests.BinSkim.Driver/BaselineTests.cs (hunk 2 above)
Comment (shaopeng-gh) [reply src/Test.FunctionalTests.BinSkim.Driver/BaselineTests.cs]: removed.

File: src/BinSkim.Driver/MultithreadedAnalyzeCommand.cs (hunk 1 above)
Comment (shaopeng-gh) [reply src/BinSkim.Driver/MultithreadedAnalyzeCommand.cs]: Please see other comments first, here can either use the current one or base one since current one incudes all.
The base one will throw null exception where 
`this.Trace` is null, so I kept the current one I have.

``` CSharp
public DefaultTraces Traces
        {
            get
            {
                defaultTraces ??=
                    this.Trace.Any()
                        ? (DefaultTraces)Enum.Parse(typeof(DefaultTraces), string.Join(",", this.Trace))
                        : DefaultTraces.None;

                return this.defaultTraces.Value;
            }
        }
```

**Comment by shaopeng-gh** [comment]
Converted to draft the new direction is to fix AnalyzeOptionsBase. It should declare ‘Traces’ as an Iset<string> not a DefaultTraces enum. Working on it.

**Comment by shaopeng-gh** [comment]
> Converted to draft the new direction is to fix AnalyzeOptionsBase. It should declare ‘Traces’ as an Iset not a DefaultTraces enum. Working on it.

Checked the SDK, the `DefaultTraces Traces` is not used I have removed: https://github.com/microsoft/sarif-sdk/pull/2676
This PR is ready to review.

**Review by michaelcfanning — APPROVED** [review]
:shipit:
